scripts/python/parthenon_metrics_app.py

```diff
@@ -0,0 +1,791 @@
+            wiki_url = wiki_url.format(
+                usr_name=self._user,
+                repo_name=self._repo_name,
+                file_name=wiki_file_name)
+            print("Wiki page url is: %s" % wiki_url)
+
+        for json_upload in json_files_to_upload:
+            self.upload(json_upload, "master", use_wiki=True)
+
+        for png_upload in png_files_to_upload:
+            self.upload(png_upload, self._default_image_branch, use_wiki=False)
+
+        self.upload(pr_wiki_page, "master", use_wiki=True)
+        if post_status:
+            self.postStatus(
+                'success',
+                commit_sha,
+                context="Parthenon Metrics App",
+                description="Performance Regression Analyzed",
+                target_url=wiki_url)
+        # 1 search for files
+        # 2 load performance metrics from wiki
+        # 3 compare the metrics
+        # 4 Create figure
+        # 5 upload figure
+        # 6 indicate pass or fail with link to figure
+
+    def checkUpToDate(self, target_branch, tests):
+        """Check to see if performance metrics for all the tests exist."""
+        super().cloneWikiRepo()
+        target_file = str(self._parthenon_wiki_dir) + "/performance_metrics_" + \
+            target_branch.replace(r'/', '-') + ".json"
+        isUpToDate = True
+        if os.path.isfile(target_file):
+            if self.branchExist(target_branch):
+                json_perf_data_parser = PerformanceDataJsonParser()
+                commit_sha = self.getLatestCommitSha(target_branch)
+                for test in tests:
+                    test_isUpToDate = json_perf_data_parser.checkDataUpToDate(
+                        target_file, target_branch, commit_sha, test)
+                    print(
+                        "Performance Metrics for test %s is uptodate: %s" %
+                        test_isUpToDate)
+                    if not test_isUpToDate:
+                        isUpToDate = False
+            else:
+                print(
+                    "Branch (%s) is not available on github." %
+                    target_branch)
+                isUpToDate = False
+            print("Performance Metrics are uptodate: %s" % isUpToDate)
+        else:
+            isUpToDate = False
+            print("Performance Metrics file is missing.")
+            print("Performance Metrics are uptodate: %s" % isUpToDate)
+
+    def printTargetBranch(self, branch):
+        target_branch = self.getBranchMergingWith(branch)
+        if target_branch is None:
+            print(
+                "Branch (%s) does not appear to not have an open pull request, no target detected." %
+                branch)
+        else:
+            print("Target branch is: %s" % target_branch)
+
+
+def main(**kwargs):
+
+    app = ParthenonApp()
+    app.initialize(
+        kwargs.pop('wiki'),
+        kwargs.pop('ignore'),
+        kwargs.pop('permissions'),
+        kwargs.pop('create'))
```

Comment:
I'm not a huge fan of `pop` here, since it deletes these values from the kwargs dict forever. Is that the intended behaviour? You could also just do `kwargs['wiki']` to get that value?

Reply:
Sounds good.